worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+        else:
+            chunksanity_random = rnd.randint(0, len(chunksanity_starting_chunks) - 1)
+            self.starting_area_item = chunksanity_starting_chunks[chunksanity_random]
+
+        # Set Starting Chunk
+        self.multiworld.push_precollected(self.create_item(self.starting_area_item))
+
+    def create_regions(self) -> None:
+        """
+        called to place player's regions into the MultiWorld's regions list. If it's hard to separate, this can be done
+        during generate_early or basic as well.
+        """
+
+        # First, create the "Menu" region to start
+        menu_region = self.create_region("Menu")
+
+        for region_row in self.region_rows:
+            self.create_region(region_row.name)
+
+        for resource_row in self.resource_rows:
+            self.create_region(resource_row.name)
+
+        # Removes the word "Area: " from the item name to get the region it applies to.
+        # I figured tacking "Area: " at the beginning would make it _easier_ to tell apart. Turns out it made it worse
+        starting_area_region = self.starting_area_item[len("Area: "):]
+        starting_entrance = menu_region.create_exit(f"Start->{starting_area_region}")
+        starting_entrance.access_rule = lambda state: state.has(self.starting_area_item, self.player)
+        starting_entrance.connect(self.region_name_to_data[starting_area_region])
+
+        # Create entrances between regions
+        for region_row in self.region_rows:
+            region = self.region_name_to_data[region_row.name]
+            for outbound_region_name in region_row.connections:
+                entrance = region.create_exit(f"{region_row.name}->{outbound_region_name.replace('*', '')}")
+                if "*" not in outbound_region_name:
+                    item_name = self.region_rows_by_name[outbound_region_name].itemReq
+                    if "*" not in item_name:
+                        entrance.access_rule = lambda state, item_name=item_name: state.has(item_name, self.player)
+                    else:
+                        self.generate_special_rules_for(entrance, region_row, outbound_region_name)
+                else:
+                    self.generate_special_rules_for(entrance, region_row, outbound_region_name)
+                entrance.connect(self.region_name_to_data[outbound_region_name.replace('*', '')])
+            for resource_region in region_row.resources:
+                if resource_region != "":
+                    entrance = region.create_exit(f"{region_row.name}->{resource_region.replace('*', '')}")
+                    if "*" not in resource_region:
+                        entrance.connect(self.region_name_to_data[resource_region])
+                    else:
+                        self.generate_special_rules_for(entrance, region_row, resource_region)
+                        entrance.connect(self.region_name_to_data[resource_region.replace('*', '')])
```

Comment:
```
for resource_region in region_row.resources:
    if not resource_region:
        continue

    entrance = region.create_exit(f"{region_row.name}->{resource_region.replace('*', '')}")

    if "*" not in resource_region:
        entrance.connect(self.region_name_to_data[resource_region])
    else:
        self.generate_special_rules_for(entrance, region_row, resource_region)
        entrance.connect(self.region_name_to_data[resource_region.replace('*', '')])
```

Again, just something I would prefer. I'm a huge indent / nesting hater. And I'm a big fan of truthiness (`if resource_region` instead of `if resource region != ""`)